routers/repo/repo.go

```diff
@@ -389,7 +392,11 @@ func Download(ctx *context.Context) {
 	}
 
 	archivePath = path.Join(archivePath, base.ShortSha(commit.ID.String())+ext)
-	if !com.IsFile(archivePath) {
+	isFile, err := util.IsFile(archivePath)
+	if err != nil {
+		log.Error("Unable to check if %s is a file. Error: %v", archivePath, err)
+	}
+	if !isFile {
```

Comment:
```if err == nil && !isExist {```

Reply:
I considered this one carefully and decided that recreating the archive was the correct thing to do here.